### ISSUE TITLE
Unexpected warning on null-coalescing operator on generic type

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#26836]
```C#
public class C<T> 
{
    T? _value = default;
    public int M() 
    {
        return _value?.ToString() == "hello" ? 1 : 2; // warning CS8602: Possible dereference of a null reference.
    }
}
```
[Example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGhhAEssmATEAagB8AsACgAAgGYABKIBMkgMIAeagD5JIgN4jJOydQD8kgPoA3KFjgRJAXklcSUbDADc23RMlsyMSQFkAFACU6sK6klqhYe4A7MZmFhD6AHTUAPYAyjAIXgDmQTa2AEQAFhA4qYWShgCMkiCSMs6SAPTNkgDuUAhkuQrpABwAbAAMMvUACqnIyGz4VvZIpBCUVqlgklCSZNhYkosQ5FRJbjoAviKnQA)

Tagging @cston. Feel free to close if this is a known issue or work item.
Found in nullable dogfood.

**Comment by jcouv** [comment]
Tested this again and we're now disallowing `T?` for unconstrained `T`.
Closing

```C#

        [Fact]
        public void TEST()
        {
            var source0 =
@"
public class C<T> 
{
    T? _value = default; // (4,5): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.
    public int M() 
    {
        return _value?.ToString() == ""hello"" ? 1 : 2; // warning CS8602: Possible dereference of a null reference.
    }
}";
            var comp0 = CreateCompilation(new[] { source0 }, options: WithNonNullTypesTrue());
            comp0.VerifyDiagnostics(
                // (4,5): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.
                //     T? _value = default;
                Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "T?").WithLocation(4, 5)
                );
        }

```
